src/pages/spawn/spawnHost/SpawnHostModal.tsx

```diff
@@ -141,22 +140,38 @@ export const SpawnHostModal: React.FC<SpawnHostModalProps> = ({
     }
   }, [awsRegions, publicKeys, dispatch]);
 
-  // Need to initialize these here so they can be used in the useEffect hook
-  let virtualWorkstationDistros = [];
-  let notVirtualWorkstationDistros = [];
+  const virtualWorkstationDistros = useMemo(
+    () =>
+      (distrosData?.distros ?? [])
+        .filter((d) => d.name.includes(distroInput))
+        .filter((d) => d.isVirtualWorkStation)
+        .sort((a, b) => a.name.localeCompare(b.name)),
+    [distrosData?.distros, distroInput]
+  );
+  const notVirtualWorkstationDistros = useMemo(
+    () =>
+      (distrosData?.distros ?? [])
+        .filter((d) => d.name.includes(distroInput))
+        .filter((d) => !d.isVirtualWorkStation)
+        .sort((a, b) => a.name.localeCompare(b.name)),
+    [distrosData?.distros, distroInput]
+  );
```

Comment:
We actually dont need to recalculate this every time. The input element automatically handles filtering the inputs. This is likely also the reason the error above also shows. Since we are changing the elements in the list every-time we enter a new character.

Reply:
This should be reverted to what we were doing before.